### PR TITLE
DAOS-2524 control: specify target count as opposed to range in config

### PR DESCRIPTION
**Pull request by tanabarr**
Specify number of targets as opposed to range of cores now we have
"first_core" index config option, add VOS targets related parameters
to the example config file as they will be used often.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

**Comment by johannlombardi** [comment]
We probably need to send an email to the internal daos-devel mailing list for this change.

File: src/control/server/config_types.go [hunk 1]
```diff
@@ -154,9 +154,9 @@ func (b *BdevClass) UnmarshalYAML(unmarshal func(interface{}) error) error {
 // server defines configuration options for DAOS IO Server instances
 type server struct {
 	Rank            *rank     `yaml:"rank"`
-	Targets         []string  `yaml:"targets"` // cpus to run xstreams
-	NrXsHelpers     int       `yaml:"nr_xs_helpers"`
-	FirstCore       int       `yaml:"first_core"`
+	Targets         int       `yaml:"targets"`       // count of cpus to run xstreams
```
Comment (liuxuezhao) [inline src/control/server/config_types.go]: minor, the // comment could be "count of storage targets per each server"

File: src/control/server/config_types.go [hunk 2]
```diff
@@ -154,9 +154,9 @@ func (b *BdevClass) UnmarshalYAML(unmarshal func(interface{}) error) error {
...
+	NrXsHelpers     int       `yaml:"nr_xs_helpers"` // offload xstreams/target
```
Comment (liuxuezhao) [inline src/control/server/config_types.go]: "count of offload/helper xstreams per each target"

File: src/control/server/config_types.go [hunk 3]
```diff
@@ -154,9 +154,9 @@ func (b *BdevClass) UnmarshalYAML(unmarshal func(interface{}) error) error {
...
+	FirstCore       int       `yaml:"first_core"`    // first cpu target
```
Comment (liuxuezhao) [inline src/control/server/config_types.go]: "offset of the first core for service xstreams"
just FYI.

**Comment by tanabarr** [comment]
@liuxuezhao adjusted as per your suggestions in the relevant config files and reference the descriptions from the source.

**Comment by johannlombardi** [comment]
Cannot be landed due to conflicts.

**Comment by jolivier23** [comment]
What needs to be announced to developers if this is landed?  They will need to update their configs accordingly?

**Comment by tanabarr** [comment]
> What needs to be announced to developers if this is landed? They will need to update their configs accordingly?

I will be announcing the internal mailing list that people either need to copy and amend an example config file or update their own local ones to specify a count value rather than a range for the "targets" parameter ("[0-7]"->"8")

**Pull request merged**